tracing/tracer.go

```diff
@@ -43,13 +43,20 @@ func init() {
 			return opentracing.ContextWithSpan(dst, opentracing.SpanFromContext(src))
 		},
 		Async: func(dst, src context.Context, next func(ctx context.Context)) {
-			span, _ := opentracing.StartSpanFromContext(
-				src,
+			span := AsSpan(opentracing.SpanFromContext(src))
+			if span == nil {
```

Comment:
`AsSpan` would never return `nil`, it's a documented feature 😅 (https://pkg.go.dev/github.com/reddit/baseplate.go/tracing#AsSpan)

but it should just work if you remove `AsSpan()`.

Reply:
I fixed this, note I still use `AsSpan` below because our `Span` implements `SpanContext` which is what `ChildOf` wants.